### PR TITLE
feat: [DHIS2-16922] Delete Tracked entity from profile Widget

### PR DESCRIPTION
**Pull request by simonadomnisoru**
[DHIS2-16922](https://dhis2.atlassian.net/browse/DHIS2-16922)

**Tech summary**
- [canCascadeDeleteTei](https://github.com/dhis2/capture-app/pull/3545/files#diff-404ad3a1fc53a164a93383521f9c7fafcbea64934b52d456e2e25e5bc8253c6cR25) - checks for the authority and caches the result using react query. 
- [canWriteData](https://github.com/dhis2/capture-app/pull/3545/files#diff-baf376fa72284493c696504764cb47837b556a0187cd5346770e41a3f4d56c30R106) - checks that the user has write access to the tracked entity type and the current program.
- The WidgetProfile exposes the `onDeleteSuccess` callback that will be used to navigate to main page by the enrollment pages.
- enable the [forceUpdateOnMount](https://github.com/dhis2/capture-app/pull/3545/files#diff-4ad92cce0db86f6818c36d5c3041b72dd390a607526e71c55a603324c3e90a22R106) logic for tracker WL. The WL will be updated when navigating to the home page so that the recently deleted tracked entity will no longer be displayed in the table.

[DHIS2-16922]: https://dhis2.atlassian.net/browse/DHIS2-16922?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/core_modules/capture-core/components/WidgetProfile/OverflowMenu/Delete/DeleteMenuItem/DeleteMenuItem.component.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    return (
+        <ConditionalTooltip content={tooltipContent} enabled={!canWriteData || !canCascadeDeleteTei}>
+            <MenuItem
+                destructive
+                dense
+                icon={<IconDelete16 />}
+                label={i18n.t('Delete {{TETName}}', {
+                    TETName: trackedEntityTypeName,
+                    interpolation: { escapeValue: false },
+                })}
```
Comment (eirikhaugstulen) [inline src/core_modules/capture-core/components/WidgetProfile/OverflowMenu/Delete/DeleteMenuItem/DeleteMenuItem.component.js:42]: Does it make sense to use a lowercased trackedEntityTypeName, similar to what we use other places?

File: src/core_modules/capture-core/components/WidgetProfile/OverflowMenu/Delete/DeleteModal/DeleteModal.componet.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+                        </ul>
+                    </NoticeBox>
+                )}
+            </ModalContent>
+            <ModalActions>
+                <ButtonStrip end>
+                    <Button onClick={() => setOpenModal(false)} secondary>
+                        {i18n.t('No, cancel')}
+                    </Button>
+                    <Button onClick={() => deleteMutation(trackedEntity)} primary loading={deleteLoading}>
```
Comment (eirikhaugstulen) [inline src/core_modules/capture-core/components/WidgetProfile/OverflowMenu/Delete/DeleteModal/DeleteModal.componet.js:54]: Should this button be destructive?

File: src/core_modules/capture-core/components/WidgetProfile/OverflowMenu/OverflowButton.js
Comment (eirikhaugstulen) [inline src/core_modules/capture-core/components/WidgetProfile/OverflowMenu/OverflowButton.js:1]: Should we still keep this file after the discussion we had in Changelogs-PR, or should we use the one in `Buttons`?

**Review by eirikhaugstulen — COMMENTED** [review]
Hey @simonadomnisoru!
Some quick comments, but otherwise great job on this! 🎉

File: src/core_modules/capture-core/components/WidgetProfile/OverflowMenu/OverflowButton.js
Comment (simonadomnisoru) [reply src/core_modules/capture-core/components/WidgetProfile/OverflowMenu/OverflowButton.js:1]: I move it into the `Buttons` folder in the latest commit. Thanks!

**Comment by simonadomnisoru** [comment]
> Hey @simonadomnisoru! Some quick comments, but otherwise great job on this! 🎉

Thanks for the feedback @eirikhaugstulen. I implemented your suggestions. Can you have a look?

**Review by JoakimSM — APPROVED** [review]
🎉

**Review by geethaalwan — APPROVED** [review]
Tested successfully on 2.41 ,2.40.4,2.39.6,2.38.7 versions

File: docs/user/using-the-capture-app.md [hunk 3]
```diff
@@ -1029,6 +1029,10 @@ Click the **Edit** button to make changes to the tracked entity instance profile
 
 ![](resources/images/enrollment-dash-tei-profile-widget-edit.png)
 
+Click the **Delete ${tracked entity type}** button to delete the tracked entity. You can confirm the action from the dialog. Once confirmed, tracked entity and all its associated enrollment and events across all programs will be deleted.
```
[the remark this answers is not in the document]
Comment (simonadomnisoru) [reply docs/user/using-the-capture-app.md]: Great suggestion. It's implemented in the latest commit. Thanks for the feedback!

**Pull request merged**